src/lib.rs

```diff
@@ -82,21 +83,16 @@ impl Report {
 
     //TODO: Use custom error to not expose xml-rs, maybe via failure
     /// Write the XML version of the Report to the given `Writer`.
-    pub fn write_xml<W: Write>(mut self, sink: W) -> writer::Result<()> {
-        // Enumerate the items in the rest suite
-        for (i,ts) in self.testsuites.iter_mut().enumerate() {
-            ts.id = i;
-        }
-
+    pub fn write_xml<W: Write>(&self, sink: W) -> writer::Result<()> {
         let mut ew = EmitterConfig::new()
             .perform_indent(true)
             .create_writer(sink);
         ew.write(XmlEvent::start_element("testsuites"))?;
 
-        for ts in &self.testsuites {
+        for (id, ts) in self.testsuites.iter().enumerate() {
```

Comment:
This is a nice solution. The only issue I have is related to the comment below about id no longer being a field that is required in the XML.